azure-pipelines.yml

```diff
@@ -5,106 +5,88 @@ trigger:
-          -ArtifactToolsList @("binskim")
-          -BinskimAdditionalRunConfigParams @("IgnorePdbLoadError < True","Recurse < True","SymbolsPath < SRV*https://msdl.microsoft.com/download/symbols")
-          -TsaInstanceURL "$(TsaInstanceURL)"
-          -TsaProjectName "$(TsaProjectName)"
-          -TsaNotificationEmail "$(TsaNotificationEmail)"
-          -TsaCodebaseAdmin "$(TsaCodebaseAdmin)"
-          -TsaBugAreaPath "$(TsaBugAreaPath)"
-          -TsaIterationPath "$(TsaIterationPath)"
-          -TsaRepositoryName "$(TsaRepositoryName)"
-          -TsaCodebaseName "$(TsaCodebaseName)"
-          -TsaPublish $True
\ No newline at end of file
+  - name: SignType
+    value: $[ coalesce(variables.OfficialSignType, 'real') ]
+  - group: core-setup-sdl-validation
+resources:
+  repositories:
+  - repository: 1ESPipelineTemplates
+    type: git
+    name: 1ESPipelineTemplates/1ESPipelineTemplates
+    ref: refs/tags/release
+extends:
+  template: v1/1ES.Official.PipelineTemplate.yml@1ESPipelineTemplates
+  parameters:
+    pool:
+      name: NetCore1ESPool-Internal
+      image: 1es-windows-2022-pt
+      os: windows
+    customBuildTags:
+    - ES365AIMigrationTooling
+    stages:
+    - stage: Build
+      jobs:
+      - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest'), eq(variables['Build.SourceBranch'], 'refs/heads/main')) }}:
+        - template: /eng/common/templates-official/job/onelocbuild.yml@self
+          parameters:
+            MirrorRepo: deployment-tools
+            LclSource: lclFilesfromPackage
+            LclPackageId: 'LCL-JUNO-PROD-MAGE'
+      - template: /eng/jobs/windows-build.yml@self
+        parameters:
+          name: win_x64
+          displayName: win-x64
+          targetArchitecture: x64
+          codeql: true
+      - template: /eng/jobs/windows-build.yml@self
+        parameters:
+          name: win_x86
+          displayName: win-x86
+          targetArchitecture: x86
+      - template: /eng/jobs/windows-build.yml@self
+        parameters:
+          name: win_arm64
+          displayName: win-arm64
+          targetArchitecture: arm64
+      - template: /eng/common/templates-official/jobs/source-build.yml@self
+        parameters:
+          platform:
+            name: 'Managed'
+            container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream9'
+      - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+        - template: /eng/common/templates-official/job/publish-build-assets.yml@self
+          parameters:
+            publishUsingPipelines: true
+            publishAssetsImmediately: true
+            dependsOn:
+            - win_x64
+            - win_x86
+            - win_arm64
+            - Source_Build_Managed
+            pool:
+              name: NetCore1ESPool-Internal
+              demands: ImageOverride -equals windows.vs2022.amd64
```

Comment:
Should this be the 1ES image? I know the docs said something about using both in certain cases.

Also can you change the name to use `$(DncEngInternalBuildPool)` for the name? I think that gets some other variable changes for free based on the docs

Reply:
I'm not sure about the image name - I've followed some other example PRs, like WindowsDesktop, which is closest to deployment-tools pipeline, as it was based on it. I'll try to change it to 1ES image.

I'll replace the pool name with variable, in both places in this yml.

Reply:
Fixed with https://github.com/dotnet/deployment-tools/pull/355/commits/14778b1a7dabe7de46b53cd4b9a057f684f3a3d7